lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,916 @@
+        """
+        query = ("SELECT n.nspname, a.rolname, n.nspacl "
+                 "FROM pg_catalog.pg_namespace AS n "
+                 "JOIN pg_authid AS a ON a.oid = n.nspowner")
+        res = self.__exec_sql(query)
+        nsp_dict = {}
+        for i in res:
+            nsp_name = i[0]
+            nsp_info = {}
+            nsp_info["nspowner"] = i[1]
+            if i[2]:
+                nsp_info["nspacl"] = i[2]
+            else:
+                nsp_info["nspacl"] = ""
+
+            nsp_dict[nsp_name] = nsp_info
+
+        self.pg_facts["namespaces"] = nsp_dict
+
+    def get_pg_version(self):
+        query = "SELECT version()"
+        raw = self.__exec_sql(query)[0][0]
+        self.pg_facts["version"] = raw.split()[1]
+
+    def get_db_info(self):
+        # Following query returns:
+        # Name, Owner, Encoding, Collate, Ctype, Access Priv, Size
+        query = ("SELECT d.datname, "
+                 "pg_catalog.pg_get_userbyid(d.datdba), "
+                 "pg_catalog.pg_encoding_to_char(d.encoding), "
+                 "d.datcollate, "
+                 "d.datctype, "
+                 "pg_catalog.array_to_string(d.datacl, E'\n'), "
+                 "CASE WHEN pg_catalog.has_database_privilege(d.datname, 'CONNECT') "
+                 "THEN pg_catalog.pg_size_pretty(pg_catalog.pg_database_size(d.datname)) "
+                 "ELSE 'No Access' END, "
+                 "t.spcname "
+                 "FROM pg_catalog.pg_database AS d "
+                 "JOIN pg_catalog.pg_tablespace t on d.dattablespace = t.oid")
+        res = self.__exec_sql(query)
+        db_dict = {}
+        for i in res:
+            db_name = i[0]
+            db_info = {}
+            db_info['owner'] = i[1]
+            db_info['encoding'] = i[2]
+            db_info['collate'] = i[3]
+            db_info['ctype'] = i[4]
+            db_info['size'] = i[6]
+            if i[5]:
+                db_info['access_priv'] = i[5]
+            else:
+                db_info['access_priv'] = ''
+
+            db_dict[db_name] = db_info
+
+        self.pg_facts["databases"] = db_dict
+
+    def __exec_sql(self, query):
+        try:
+            self.cursor.execute(query)
+            res = self.cursor.fetchall()
+            if res:
+                return res
+        except SQLParseError as e:
+            self.module.fail_json(msg=to_native(e),
+                                  exception=traceback.format_exc())
+            self.cursor.close()
+        except psycopg2.ProgrammingError as e:
+            self.module.fail_json(msg="Cannot execute SQL '%s': %s" % (query, to_native(e)),
+                                  exception=traceback.format_exc())
+            self.cursor.close()
+        return False
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    argument_spec = pgutils.postgres_common_argument_spec()
+    argument_spec.update(dict(
+        db=dict(default='', type=str),
```

Comment:
Beware, it needs to be `type='str'`. Please always put the `type` option at the start if your dict, it's easier to read the argspec (as every parameter must have a type-definition).
Usually we do not set defaults so you can easily detect if the value was set, or not (None).
```suggestion
        db=dict(type='str'),
```